### PR TITLE
Digital post office

### PR DESCRIPTION
**Pull request by rimi-itk**
https://jira.itkdev.dk/browse/TVIST1-432

Adds display of digital post (by hijacking the “communication” route).

### CODE REVIEW

File: templates/case/communication/digital_post/index.html.twig [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+{% block content %}
+    <table class="table table-striped">
+        <thead>
+            <tr>
+                <th>{% trans %}Status{% endtrans %}</th>
+                <th>{% trans %}Created at{% endtrans %}</th>
+                <th>{% trans %}Sent at{% endtrans %}</th>
+                <th>{% trans %}Document{% endtrans %}</th>
+                <th class="number">{% trans %}Number of attachments{% endtrans %}</th>
+                <th></th>
```
Comment (jekuaitk) [inline templates/case/communication/digital_post/index.html.twig:24]: It occurred to me that we don't show recipients anywhere. Consider adding it here or at least in `digital_post_show`.

File: templates/case/communication/digital_post/show.html.twig [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                <th>{% trans %}Sent at{% endtrans %}</th>
+                <td>{{ digital_post.sentAt ? digital_post.sentAt|date('Y-m-d H:i:s') : '–' }}</td>
+            </tr>
+            <tr>
+                <th>{% trans %}Document{% endtrans %}</th>
+                <td>
+                    {{ digital_post.document.documentName }}
+                    {{ _self.show_document_content_button(case, digital_post.document) }}
+                </td>
+            </tr>
```
Comment (jekuaitk) [inline templates/case/communication/digital_post/show.html.twig:43]: Consider also showing recipients here.

**Review by jekuaitk — CHANGES_REQUESTED** [review]

File: templates/case/communication/digital_post/index.html.twig (hunk 1 above)
Comment (rimi-itk) [reply templates/case/communication/digital_post/index.html.twig]: I've added it to the details view.

File: templates/case/communication/digital_post/show.html.twig [hunk 3]
```diff
@@ -16,51 +16,63 @@
...
+            <dt class="col-sm-3">{% trans %}Created at{% endtrans %}</dt>
+            <dd class="col-sm-9">{{ digital_post.createdAt ? digital_post.createdAt|date('Y-m-d H:i:s') : '–' }}</dd>
+
+            <dt class="col-sm-3">{% trans %}Status{% endtrans %}</dt>
+            <dd class="col-sm-9">{{ digital_post.status ?? '–' }}</dd>
+
+            <dt class="col-sm-3">{% trans %}Sent at{% endtrans %}</dt>
+            <dd class="col-sm-9">{{ digital_post.sentAt ? digital_post.sentAt|date('Y-m-d H:i:s') : '–' }}</dd>
+
+            <dt class="col-sm-3">{% trans %}Recipients{% endtrans %}</dt>
```
Comment (jekuaitk) [inline templates/case/communication/digital_post/show.html.twig:36]: Missing a translation.

File: templates/case/communication/digital_post/show.html.twig [hunk 4]
```diff
@@ -16,51 +16,63 @@
...
+{% block javascripts %}
+    {{ parent() }}
+    <script>
+        // Compute height if iframe element to make it take up (most of) the remaining space on the page.
+        window.addEventListener('load', function () {
+            const iframe = document.querySelector('[name="document-content"]');
+            const rect = iframe.getBoundingClientRect()
+            iframe.style.height = (window.innerHeight-rect.top-40)+'px'
+        })
+    </script>
```
Comment (jekuaitk) [inline templates/case/communication/digital_post/show.html.twig:77]: Nice!

**Review by jekuaitk — CHANGES_REQUESTED** [review]

**Review by jekuaitk — APPROVED** [review]

**Pull request merged**